refract-spec.md

```diff
@@ -12,35 +12,46 @@ This document conforms to RFC 2119, which says:
 
 [MSON](https://github.com/apiaryio/mson) is used throughout this document to describe and define data structures.
 
-## Data Structure
+## Base Element
 
 Refract provides a single structure for data, which will be referred to throughout this document as an element. All elements found in a Refract document SHOULD conform to the element structure as defined here.
 
-### Element
+### Element (object)
 
-The Refract Element contains three properties, `element`, `attributes`, and `content`, as defined below. This Element can be used recursively throughout the document, even as a value for each of its own properties.
+The Refract Element contains four properties: `element`, `meta`, `attributes`, and `content`, as defined below. This Element MAY be used recursively throughout the document, even as a value for each of its own properties.
 
 #### Properties
 
-- `element` (enum, required)
+- `element` (string, required)
 
-    The `element` property defines the type of element. It MAY be a string that references an element type, or it MAY be an `element` itself that provides additional data about the element type.
+    The `element` property defines the name of element. It MUST be a string that references an element, which SHOULD be defined.
 
-    - Members
-        - (string)
-        - (Element)
+- `meta` (enum)
+
+  The `meta` property is a reserved object for Refract-specific values.
+
+  - Members
+      - (object)
+          - `id` - Unique Identifier, MUST be unique throughout the document
+          - `ref` (Link) - Link to referenced element or type
+          - `class` (array[string]) - Array of classifications for given element
+          - `prefix` (string) - Prefix in which element MAY be found
+          - `namespaces` (array[Link]) - Include elements from given namespaces or prefix elements from given namespace
+          - `title` (string) - Human-readable title of element
+          - `description` (string) - Human-readable description of element
+      - (array[Member Element])
 
 - `attributes` (enum)
 
-    The `attributes` property defines attributes about the given instance of the element, as specified by the `element` property. It MAY be an array of elements, or it MAY be an object. The keys of the object SHOULD be defined element types with valid values for the corresponding element types.
+    The `attributes` property defines attributes about the given instance of the element, as specified by the `element` property.
 
     - Members
-        - (object)
-        - array[Element]
+        - (string)
```

Comment:
I believe this should be `(object)` (a hash) not a string – attributes is a hash or array of elements vs. attributes is a string or array of elements